### PR TITLE
Scope of Strimzi

### PR DESCRIPTION
**Pull request by sjwoodman**
Document to begin discussion about scope of Strimzi and what's more and less suitable for inclusion

### CODE REVIEW

File: INCLUSION.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#Inclusion 
+
+Strimzi welcomes contributions from the community both in code and defining the direction of the project.
+There are no concrete rules for what features should or should not be added to Strimzi but we expect the community to guide this.
+
+The goal of Strimzi is to enable the Running and operating Kafka on Kubernetes effectively.
```
Comment (kyguy) [inline INCLUSION.md]: Running -> running

Comment (kyguy) [reply INCLUSION.md]: running of Kafka

File: INCLUSION.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Strimzi welcomes contributions from the community both in code and defining the direction of the project.
+There are no concrete rules for what features should or should not be added to Strimzi but we expect the community to guide this.
+
+The goal of Strimzi is to enable the running and operating of Apache Kafka on Kubernetes effectively. 
+
+Suggested features that would likely be of interest to Strimzi:
+
+* Operators and/or container images for Apache Kafka components.
+* Support and/or documentation for additional kubernetes distributions.
```
Comment (ppatierno) [inline INCLUSION.md]:
```suggestion
* Support and/or documentation for additional Kubernetes distributions.
```

File: INCLUSION.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Suggested features that may be out of scope of Strimzi:
+
+* Additional language clients and Kafka quick starts - the upstream Apache Kafka project is more suitable.
+* Data streaming platforms.
+* Vendor specific integrations especially those with a single implementation.
+
+For some features the decision may be influenced by how well the feature integrates with Kubernetes.
+An example could be a console for monitoring Kafka.
+If this were kubernetes aware then it would be more suitable for Strimzi than if it was not.
```
Comment (ppatierno) [inline INCLUSION.md]:
```suggestion
If this were Kubernetes aware then it would be more suitable for Strimzi than if it was not.  
```

**Comment by emedina** [comment]
I would like to see the Schema Registry included, as this is typically a component required in most of Kafka deployments.

**Comment by tombentley** [comment]
While I don't object to anything in the `INCLUSION.md` _per se_, I find it too woolly to really be useful. It doesn't describe any principles or processes. I think it's very hard to write down any meaningful principles about what would or would not be considered for inclusion in Strimzi, because ultimately they're all going to be judged on a case-by-case basis. 

What I think would be more useful for the community would be a description of the process we would use in making that judgement. We need a process that recognises that

* simple bug fixes should have a streamlined route to inclusion.
* people wanting to contribute bigger features often want/need up-front guidance about how to approach the problem in a way which is most likely to result in their efforts being merged. I'm not saying people should necessarily seek permission to open a PR, but often they don't want to waste a lot of time working on something which might get rejected. 
* if/when the maintainers decide that a feature is not appropriate for inclusion that the reasons why not are communicated clearly.
* non-maintainers should have a way of influencing the direction of the project's priorities. For this I mean a way to say "I can't offer to help on this, but it's a feature I'd benefit from". This wouldn't be binding on the maintainers, but would be something they should take into consideration in their treatment of the feature.

Until now we've more-or-less relied on people coming forward and saying "I'd like this feature and am willing to contribute the work necessary". I think it would be better if we just publish/advertise the on-ramp to the process, so people don't have to ask, and have a clear understanding before they do of what the process is.

As an example, Kafka's KIP process is easily discovered by potential contributors and makes their process clear. I'm not necessarily suggesting we use that for Strimzi but I do think giving people visibility of the process before they feel the need to ask would be a good step.

WDYT @ppatierno @scholzj ?

**Comment by ppatierno** [comment]
I think that it's not formalized but we already have this process in place somehow.
A lot of users just open issues asking for a feature that could be useful for them and maintainers engage di discussion to understand with such a users if it really fits in the project. Sometimes the users propose themselves to do the work and opening a PR. I don't think that until now the maintainers refused PRs and users wasted their time on implementing something which doesn't fit in the project.
Regarding the comparison with KIPs even, in that case, there are drawbacks: you open and write a KIP starts the discussion and then it's refused by voting. Writing a KIP doesn't ensure that your work will be accepted. I see it just like a GitHub issue where maintainers and users can discuss.
Of course, I agree that it's not well formalized as a KIP is.

Btw, I agree that we should make clear the process for making contributions.

**Comment by marcospassos** [comment]
The last missing piece is Kafka Registry, any chances of adding support anytime soon?

**Comment by sjwoodman** [comment]
Yes, fine by me.

**Pull request closed without merging**